### PR TITLE
Fix FEWS NET clobber

### PR DESCRIPTION
**Pull request by turnerm**
Great catch @caldwellst that clobber was not implemented properly. I've tried to fix it and also add a test.

I'm wondering if we should always be testing our clobber implementations... seems like a  huge pain though.

The changes are `raster.py` are because mypy suddenly started complaining!

### CODE REVIEW

File: src/aatoolbox/datasources/fewsnet/fewsnet.py [hunk 1]
```diff
@@ -371,35 +380,37 @@ def _download_zip(
...
         )
         if country_dir.is_dir():
             return country_dir
         elif region_dir.is_dir():
             return region_dir
 
         raise FileNotFoundError(
-            f"No data found for {pub_year}-{pub_month} covering "
+            f"No data found for {pub_year}-{pub_month_str} covering "
             "{self._country_config.iso3} "
```
Comment (caldwellst) [inline src/aatoolbox/datasources/fewsnet/fewsnet.py:405]: This and one below should be f-strings.

File: src/aatoolbox/datasources/fewsnet/fewsnet.py [hunk 2]
```diff
@@ -209,17 +209,17 @@ def load(  # type: ignore
...
             dir_path=dir_path, projection_period=projection_period
         )
         return gpd.read_file(file_path)
 
     @staticmethod
-    def _check_date_validity(pub_year, pub_month):
+    def _check_date_validity(pub_year: int, pub_month: int):
         try:
             pub_date = datetime.datetime(year=pub_year, month=pub_month, day=1)
         except ValueError as err:
```
Comment (caldwellst) [inline src/aatoolbox/datasources/fewsnet/fewsnet.py:225]: Line 227 below this should be an f-string.

Comment (caldwellst) [reply src/aatoolbox/datasources/fewsnet/fewsnet.py:225]: Also line 233

**Review by caldwellst — APPROVED** [review]
All looks good! Was wondering about the testing. I was wondering, since the `@check_file_existence` decorator is always used on internal functions/methods, at least so far, could we require that `clobber` is passed, and raise a key error if not, so that if functions don't use clobber then in the internal method they can explicitly pass `True` or `False` even if `clobber` isn't an argument in the parent function?

File: src/aatoolbox/datasources/fewsnet/fewsnet.py (hunk 1 above)
Comment (turnerm) [reply src/aatoolbox/datasources/fewsnet/fewsnet.py:405]: Fixed!

**Comment by turnerm** [comment]
@caldwellst great idea to make clobber required!

**Review by caldwellst — APPROVED** [review]

**Pull request merged**